engine/realtimelogger.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <concurrentqueue.h>
+
+#include <fmt/core.h>
+#include <thread>
+
+namespace pong
+{
+
+static constexpr int MAX_LOG_MESSAGE_SIZE = 512;
+
+struct LogData
+{
+    int mLevel { 0 };
+    int mSequenceNum { 0 };
+    char mMessage[MAX_LOG_MESSAGE_SIZE] {};
+};
+
+class RealTimeLogger
+{
+private:
+    static constexpr int MAX_LOGS = 100;
+
+    std::thread mThread {};
```

Comment:
consider using a scoped thread here and avoid having to have a destructor/constructor.

Alternatively, write or delete the move/copy constructors. If someone tries to copy/move right now you will get UB and a likely crash

Reply:
after looking at the mAlive thing, I think you likely still need a destructor, (unless you want to get really crazy and write some kind of `ScopedTrueBool`

So what you could do here is:

1. Use a jthread
2. AND delete copy/move constructors


a reminder from rule of five "if you define one of them, you must define all of them". So by making this class have a destructor, you must write the rest of them.

Dave had an interview question literally two days ago from Apple where the answer was to complete the rule of five. Keep all this in mind :)

Reply:
So I kept the constructor and deleted the copy/move constructors. It seems that jthread is in C++20 though. I may want to move to C++20 at some point, but likely not now. I was very curious if I can use coroutines for callback instead of the timer class from earlier, so still open to trying eventually. Plus I tried to switch to C++20 and got some build errors so it will take some looking into lol.

Reply:
ah, cool!! Didn't realize it was the bleeding edge :) cppcon cooked my brain too many times, making me think I'm living in the future